Lib/idlelib/stackviewer.py

```diff
@@ -1,33 +1,31 @@
 import linecache
 import os
-import sys
 
 import tkinter as tk
 
 from idlelib.debugobj import ObjectTreeItem, make_objecttreeitem
 from idlelib.tree import TreeNode, TreeItem, ScrolledCanvas
 
-def StackBrowser(root, flist=None, tb=None, top=None):
+def StackBrowser(root, flist=None, exc=None, top=None):
```

Comment:
flist can be None because optional use in double_click below is guarded.
tb could be None because replaced with sys.last_traceback in StackTreeItem.get_stack.
replacement exc must be passed, hence should be second param here and in all calls.
```suggestion
def stackbrowser(root, exc, flist=None, top=None):
```
The call below has a suggested replacement.
Replacement for call in test_stackviewer is in file comment.

The 3rd call is in def open_stack_viewer, line 1366 in pyshell.  It is only used in -n, no subprocess node.
Before this PRis merged, `sys.last_traceback` should be changed to `exc := sys.last_value` (or last_exc) in a separate  pair of PRs (one each for 3.11 and 3.12) that will allow this PR to be backported as is.

Reply:
Fixed def, test call, htest call, pyshell call.